workflow/metrics/operators.go

```diff
@@ -0,0 +1,109 @@
+package metrics
+
+import (
+	"context"
+	"fmt"
+
+	log "github.com/sirupsen/logrus"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/metric"
+)
+
+func (m *Metrics) addInt(ctx context.Context, name string, val int64, labels instAttribs) {
+	if instrument, ok := m.allInstruments[name]; ok {
+		instrument.addInt(ctx, val, labels)
+	} else {
+		log.Errorf("Metrics addInt() to non-existent metric %s", name)
```

Comment:
we should never actually hit these else blocks if we use the wrapper functions that populate the key `name`.

Reply:
Yeah, we should never hit them. I'd like to leave them in for all those times when should didn't happen.

Reply:
Sure that makes sense